src/evaluator/functions.ts

```diff
@@ -456,11 +456,47 @@ diff.changedOnly = () => {
 }
 diff.changedOnly.arity = 3
 
+const array: FunctionSet = {}
+
+array.unique = async function (args, scope, execute) {
+  if (args.length > 1) {
+    throw new Error('Incorrect number of arguments to function "array::unique"')
+  }
+
+  const value = await execute(args[0], scope)
+  if (!value.isArray()) {
+    return NULL_VALUE
+  }
+
+  const result = []
+  const added = new Set()
+  for await (const iter of value) {
+    switch (iter.type) {
+      case 'number':
+      case 'string':
+      case 'boolean':
+      case 'datetime':
+        if (!added.has(iter.data)) {
+          added.add(iter.data)
+
+          result.push(iter.data)
+        }
+        break
+      default:
+        const element = await iter.get()
+        result.push(element)
+    }
+  }
+
+  return fromJS(result)
```

Comment:
It's possible to implement this in a streaming manner:

```suggestion
  return new new StreamValue(async function* () {
    const added = new Set()
    for await (const iter of value) {
      switch (iter.type) {
        case 'number':
        case 'string':
        case 'boolean':
        case 'datetime':
          if (!added.has(iter.data)) {
            added.add(iter.data)
            yield iter
          }
          break
        default:
          yield iter
      }
    }
  })
```